app/controllers/api/v0/manifests_controller.rb

```diff
@@ -9,7 +9,34 @@ def search
     end
   end
 
+  def validate
+    begin
+      manifest_content = JSON.parse(request.body.read)
+    rescue JSON::ParserError => err
+      render json: {message: "Invalid JSON in request: #{err}"}, status: 400
+      return
+    end
+
+    validator = ManifestValidator.new(manifest_content)
+    if validator.run
+      render json: {message: "Manifest structure is valid"}, status: 200
+    else
+      render json: {
+        message: "Validation failed",
+        errors: validator.error_messages
+      }.to_json, status: 422
+    end
+  end
+
   def create
+    validator = ManifestValidator.new(manifest_params)
+    unless validator.run
+      render json: {
+        message: "Validation failed",
+        errors: validator.error_messages
+      }.to_json, status: 422
+      return
```

Comment:
ahh just viewed whole file and now I see why we have this.

what about doing an `if` and `else` in this `create` method rather than an `unless` with a return. I think that might make the logic easier to understand